Update release documents (#282)
Babylonpartners/ios-playbook: Cookbook/Technical-Documents/ReleaseProcess.md

## 2. Release step-by-step

### Phase 1: Pre-release considerations

> 1. To help expedience of the release, any required feature flag changes should be done prior to the 2am cut-off on the Friday. If this should be missed, a concession can be made so that the changes will be allowed to be made during the Friday. However, this should not be a regular occurance as it requires intervention from the assigned release engineers to create new builds.

Reviewer: ```suggestion
1. To help expedite the release, any required feature flag changes should be done prior to the 2am cut-off on the Friday. If these are missed, a concession can be made so that the changes will be allowed to be made during the Friday. However, this should not be a regular occurrence as it requires intervention from the assigned release engineers to create new builds.
```

  1. On the sidebar click `+ Version or Platform` and select `iOS`.
  1. Input the new version number.

### Phase 3: Test and fix bugs
*It starts after the App Center build has been delivered and it can take several cycles*

1. Testers will then begin their work against the build you just created.

> 1. Any hotfix should target the `develop` branch first. These PRs need to be reviewed not just by the relevant squad or platform QA (Bibin Paul), but also by one of the release engineers assigned to the release. This is to ensure visibility of changes being requested to be made, and to ensure the correct builds are available for validation.

Reviewer: ```suggestion
1. Any hotfix should target the `develop` branch first. 
1. These PRs need to be reviewed by the relevant squad or platform QA (Bibin Paul) & one of the release engineers assigned to the release. This is to ensure visibility of changes being requested to be made, and to ensure the correct builds are available for validation.
```

> 1. Any hotfix should target the `develop` branch first. These PRs need to be reviewed not just by the relevant squad or platform QA (Bibin Paul), but also by one of the release engineers assigned to the release. This is to ensure visibility of changes being requested to be made, and to ensure the correct builds are available for validation.

Reviewer: Is there a link to a page which keeps current platform QA up-to-date? Instead of using Bibin Paul's name directly here.

Author: Hmm no idea if we've such a page. In any case the document already has Andreea and Lilla's names there, explicitely stated.

> 1. Any hotfix should target the `develop` branch first. These PRs need to be reviewed not just by the relevant squad or platform QA (Bibin Paul), but also by one of the release engineers assigned to the release. This is to ensure visibility of changes being requested to be made, and to ensure the correct builds are available for validation.

Reviewer: We also need to add the point -  * The exception being that if an issue is not reproducible on develop , then release first has to be done. But it should be fully understand why it is not reproducible to help ensure there is no regression for the next release

https://babylonhealth.slack.com/archives/GNYBQ06P5/p1573728512152900?thread_ts=1573724409.144400&cid=GNYBQ06P5

  1. Input the new version number.

### Phase 3: Test and fix bugs
*It starts after the App Center build has been delivered and it can take several cycles*

1. Testers will then begin their work against the build you just created.
1. Any hotfix should target the `develop` branch first. 
1. These PRs need to be reviewed by the relevant squad or platform QA (Bibin Paul) & one of the release engineers assigned to the release. This is to ensure visibility of changes being requested to be made, and to ensure the correct builds are available for validation. The exception being that if an issue is not reproducible on `develop` then we can merge directly with `release`. Still, it should be fully understandable why it is not reproducible to help ensure there is no regression for the next release.
    * Bear in mind that two approvals from other engineers assigned by Pull Assigners is not enough in this particular case.
    * After merging, we'll need to cherry-pick the commit via a PR to the release branch. Doing this will ensure that both branches are up to date and reduces the risk of conflicts when merging the release branch back to develop. 

>     * Furthermore, the issue for the hotfix has to be added to the release JIRA board.

Reviewer 2: We can explain a hotfix JIRA ticket is automatically added to the release JIRA board by setting the release number in `Fix version`.

>     * Furthermore, the issue for the hotfix has to be added to the release JIRA board.

Author: ```suggestion
    * Furthermore, the issue for the hotfix has to be added to the release JIRA board. This can be done automatically by setting the release number in `Fix version` in the hotfix's JIRA card.
```

Like this @AnnKatF ?

Reviewer 2: Perfect!
